src/any_llm/providers/azure/azure.py

```diff
@@ -87,6 +123,47 @@ def _stream_completion(
         for chunk in azure_stream:
             yield _create_openai_chunk_from_azure_chunk(chunk)
 
+    async def acompletion(
+        self,
+        params: CompletionParams,
+        **kwargs: Any,
+    ) -> ChatCompletion | AsyncIterator[ChatCompletionChunk]:
+        """Create a chat completion using Azure AI Inference SDK."""
+        client: aio.ChatCompletionsClient = self._create_chat_client_async()
+
+        if params.reasoning_effort == "auto":
+            params.reasoning_effort = None
+
+        azure_response_format = None
+        if params.response_format:
+            azure_response_format = _convert_response_format(params.response_format)
+
+        call_kwargs = params.model_dump(exclude_none=True, exclude={"model_id", "messages", "response_format"})
+        if params.stream:
+            if azure_response_format:
+                call_kwargs["response_format"] = azure_response_format
```

Comment:
Can this be factored up out of the `if params.stream` ? why have it both here and line 152?